contracts/TokenManager.sol

```diff
@@ -221,16 +258,32 @@ contract TokenManager is ITokenController, IForwarder, AragonApp {
     * @return False if the controller does not authorize the transfer
     */
     function onTransfer(address _from, address _to, uint256 _amount) external onlyToken returns (bool) {
-        return _isBalanceIncreaseAllowed(_to, _amount) && _transferableBalance(_from, getTimestamp()) >= _amount;
+        bool transferable = _isBalanceIncreaseAllowed(_to, _amount) && _transferableBalance(_from, getTimestamp()) >= _amount;
+        uint256 i = 0;
+        while (transferable && i < hooksLength) {
+            if (address(hooks[i]) != 0) {
+                transferable = transferable && hooks[i].onTransfer(_from, _to, _amount);
+            }
+            i++;
+        }
+        return transferable;
     }
 
     /**
     * @dev Notifies the controller about an approval allowing the controller to react if desired
     *      Initialization check is implicitly provided by `onlyToken()`.
     * @return False if the controller does not authorize the approval
     */
-    function onApprove(address, address, uint) external onlyToken returns (bool) {
-        return true;
+    function onApprove(address _holder, address _spender, uint _amount) external onlyToken returns (bool) {
+        bool approved = true;
+        uint256 i = 0;
+        while (approved && i < hooksLength) {
+            if (address(hooks[i]) != 0) {
+                approved = approved && hooks[i].onApprove(_holder, _spender, _amount);
```

Comment:
As suggested previously, isn't `approved` always true here?

Reply:
The same as above, it is an AND boolean accumulator, and it is false as soon as one hook returns false.

Reply:
Can you explain what you mean by boolean "accumulator"? Typically I consider this a boolean operator.

Reply:
My mistake, those variables are always true. I was using that in a `for` before, and this is why I was using `approved` in that way. In a `while` it is not needed anymore 😅.